src/FSAL/FSAL_CORTXFS/mds.c

```diff
@@ -269,25 +280,146 @@ int kvsfs_pmds_ini(struct kvsfs_fsal_module *kvsfs,
 		goto out;
 	}
 
-	mds_ctx->pnfs_role = KVSFS_PNFS_DISABLED;
-	mds_ctx->mds = gsh_calloc(1, sizeof(struct kvsfs_pnfs_mds_info));
-	if (mds_ctx->mds == NULL) {
-		LogCrit(COMPONENT_PNFS,
+
+	/** 
+	 * TODO: When the pNFS config is moved to cortx.conf file, process
+	 * kvsfs_params here and update kvsfs->pnfs_role and
+	 * kvsfs->mds_ctx. Until then, just initialize kvsfs->mds_ctx.
+	 * In future, initialize only if pnfs_role > KVSFS_PNFS_DS.
+	 */
+	/* Read the pNFS.conf file */
+	rc = config_from_file("libcortxfs", CONF_FILE, &cfg_items,INI_STOP_ON_ERROR, &errors);
+	if(rc) {
+			free_ini_config_errors(errors);
+			rc = -rc;
+	} 
+
+	/* Read the configuration parameters */
+	rc = get_config_item("pNFS","pNFS_role",cfg_items, &items);
+	if(items==NULL) {
+		LogCrit(COMPONENT_PNFS,"pNFS parameters configured incorrectly\n");
+		rc = -rc;
+		goto out; 
+	}
+	pNFS_role = get_string_config_value(items, NULL);
+	if(strcmp(pNFS_role,"MDS") == 0) {
+		mds_ctx->pnfs_role = KVSFS_PNFS_MDS;
+	} 
+	else if(strcmp(pNFS_role,"DS") == 0) { 
+		mds_ctx->pnfs_role = KVSFS_PNFS_DS;
+	}
+	else if(strcmp(pNFS_role,"BOTH") == 0)	{
+		mds_ctx->pnfs_role = KVSFS_PNFS_BOTH;
+	}
+	else {
+		mds_ctx->pnfs_role = KVSFS_PNFS_DISABLED;
+	}
+	
+	
+	if(mds_ctx->pnfs_role > KVSFS_PNFS_DS )
+	{	
+		mds_ctx->mds = gsh_calloc(1, sizeof(struct kvsfs_pnfs_mds_info));
+		if (mds_ctx->mds == NULL) {
+			LogCrit(COMPONENT_PNFS,
 			"gsh_calloc failed: %u", sizeof(*mds_ctx->mds));
-		rc = -ENOMEM;
-		goto out;
+			rc = -ENOMEM;
+			goto out;
+		}
+		mds_ctx->mds->stripe_unit = KVSFS_PNFS_STRIPE_UNIT;	
+	
+		rc = get_config_item("pNFS","NumDs",cfg_items, &items);
+		if ( items == NULL){
```

Comment:
if the role is BOTH, then do we still need explicit NumDs ? I suppose it may not have any value & we should consider it as 0

Reply:
Yes, Both just implies that the MDS will also be setup as DS.But there will be other DS as well.So, NumDs=DS in cluster+MDS that is setup as DS